src/Diagnostics.Logger/ETWProvider/DiagnosticsETWProvider.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Text;
+
+namespace Diagnostics.Logger
+{
+    [EventSource(Name = "Microsoft-Azure-AppService-Diagnostics")]
+    public sealed class DiagnosticsETWProvider : DiagnosticsEventSourceBase
+    {
+        public static DiagnosticsETWProvider Instance = new DiagnosticsETWProvider();
+
+        #region Compile Host Events (ID Range : 1000 - 1999)
+
+        [Event(1000, Level = EventLevel.Informational, Channel = EventChannel.Admin, Message = ETWMessageTemplates.LogCompilerHostMessage)]
+        public void LogCompilerHostMessage(string Message)
+        {
+            WriteDiagnosticsEvent(1000, Message);
+        }
+
+        [Event(1001, Level = EventLevel.Error, Channel = EventChannel.Admin, Message = ETWMessageTemplates.LogCompilerHostUnhandledException)]
+        public void LogCompilerHostUnhandledException(string RequestId, string Source, string ExceptionType, string ExceptionDetails)
+        {
+            WriteDiagnosticsEvent(1001,
+                RequestId,
+                Source,
+                ExceptionType,
+                ExceptionDetails);
+        }
+
+        [Event(1002, Level = EventLevel.Informational, Channel = EventChannel.Admin, Message = ETWMessageTemplates.LogCompilerHostAPISummary)]
+        public void LogCompilerHostAPISummary(string RequestId, string Address, string Verb, int StatusCode, long LatencyInMilliseconds, string StartTime, string EndTime)
```

Comment:
We should also log the Id of the detector id specified in the csx file. I can see near future value in tracking that kind of telemetry.

Reply:
for that, you can use request id as that is getting passed

Reply:
It would be difficult to derive useful insights about a detector with the requestid since there is a unique requestid for each compilation.

 It might be helpful to have detectorId so that you can pivot on that and get useful telemetry. For example: average amount time roselyn has been taking to compile this detector in the past 7 days. That information could be useful for us to optimize our platform.

Reply:
for compiler API request, there is no detector Id. The input is a general script csx file.  
Those kind of insights can easily be drawn doing a join between 2002(detector api event) and 1002(compiler api event)

Reply:
I see so I would do a join on both of those tables and look for the detector name in the address is the way to do it right?

Reply:
See the compilation is not happening for every detector (since the assemblies are already compiled and saved). Compilation only happens for one API (/query). So that insight you mentioned will not be useful.
What you want to know would be what's the average compilation time for script, how many times the memory threshold is breached (from my experiements, I haven't seen  any variations in compilation time per detector)